modules/ROOT/pages/tips-and-tricks.adoc

```diff
@@ -160,3 +160,20 @@ or setup shell `alias`es as needed to make them available to the CLI like so:
   $ alias evince="flatpak run org.gnome.Evince"
     # or alias evince="org.gnome.Evince"
   $ evince
+
+== Enabling TPM2 for LUKS 
+
+It is possible to use the TPM2 device in your machine to automatically decrypt your LUKS-encrypted device. It is important to note a few things regarding this setup: first, it is trading security for convenience. Anyone who has access to both the disk and the TPM2 device can unlock your data. Second, it is crucial to keep a LUKS passphrase in addition to the TPM2 device that you can use to unlock the LUKS device; if you update BIOS, or the kernel shim, or make other substantial changes to your device, the TPM will refuse to sign your session and you will fall back to manually entering a LUKS passphrase. In this case, you can re-enable TPM2 for unlocking by re-running the enroll command below. This tip assumes you have a single LUKS device created during the Anaconda installer; if you have more than one LUKS device, you will need to correctly identify the device you want to enroll. 
+
+To set up TPM2 unlocking, first, find the LUKS device you want to enroll. This is probably in `/etc/crypttab`. You can also use `sudo cryptsetup status /dev/mapper/luks*` to identify the device.
+
+Next, enable the required initramfs and kernel features. Note that the initramfs command below will overwrite any other initramfs changes you have made: 
+
+  $ sudo rpm-ostree kargs --append=rd.luks.options=tpm2-device=auto
+  $ sudo rpm-ostree initramfs --enable --arg=-a --arg=systemd-pcrphase
+
+Then, using the device you identified with 'crpysetup status' previously, enroll the device: 
+
+  $ sudo systemd-cryptenroll --tpm2-device=auto --tpm2-pcrs=0+7 /dev/boot-device
```

Comment:
Is 0+7 secure enough? Correct me if I am wrong, but changing the kernel parameters would allow to mount the disk and change the root password.

Reply:
It's not, but it's the bset we can do right now as the kernel parameters change on every boot as ostree includes the hash of the deployment to boot in the kernel parameters.

Reply:
I don't think it's a good idea to recommend this because it will appear to work for a while, until such time as you update shim, and then it will break.

We discovered this in e.g. https://github.com/containers/bootc/issues/421#issuecomment-2062771415

At least if you have this it needs to have a big warning label about this...

Reply:
Yes, the fact that the TPM2-backed credential can stop working is well understood and has been discussed a few times in the diff discussions in this PR already

Reply:
Also, I think the warning should be sufficient but some documentation *should* be merged letting folks know how to use this. `shim` updates seem to be about as regular as UEFI platform f/w updates from my perspective. I boot my computer daily, so I'm certainly glad I only have to type my password once every 500-1000 boots nowadays, rather than every time :sweat_smile: